### PR TITLE
response_content_disposition for S3 uploads

### LINKED ISSUE #7309: S3 .carto file download is renamed to .zip
Visualization exports using S3 download the file with .zip extension instead of .carto. It's not a problem for importing to work but it'd be nicer keeping the original extension.

cc @gfiorav

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #7309.

@gfiorav CR this, please.

### CODE REVIEW

File: app/helpers/file_upload.rb [hunk 1]
```diff
@@ -18,25 +18,29 @@ def get_uploads_path
...
-          filepath = params[:filename].to_s
+          filepath = filename_param.to_s
           load_file_from_request_body = true
         end
-      when params[:file].present?
-        filename = params[:file].original_filename rescue params[:file].to_s
-        filepath = params[:file].path rescue ''
+      when file_param.present?
+        filename = file_param.original_filename rescue file_param.to_s
+        filepath = file_param.path rescue ''
```
Comment (gfiorav) [inline app/helpers/file_upload.rb]: Agreed with all complains about modifier for of `rescue`. Totally ignorable though.

**Comment by gfiorav** [comment]
📦 for the most part. It would be nice if we had a test to verify content disposition params are generated.

Nice stuff!

**Comment by juanignaciosl** [comment]
@gfiorav I promise that I've given a thought about adding a test about the content disposition params, but file upload hasn't a test suite right now and adding it is too expensive for this change. The same happens with the `rescue` refactor: a lot of file upload helper code depends on that and there's no code coverage, for it, so I prefer not touching it :(

**Comment by gfiorav** [comment]
🌴

**Pull request merged**